app/controllers/users_controller.rb

```diff
@@ -110,15 +110,16 @@ def onboarding_update
 
   def join_org
     authorize User
-    if (@organization = Organization.find_by(secret: params[:org_secret]))
+    if (@organization = Organization.find_by(secret: "params[:org_secret].strip"))
       ActiveRecord::Base.transaction do
         current_user.update(organization_id: @organization.id)
         OrganizationMembership.create(user_id: current_user.id, organization_id: current_user.organization_id, type_of_user: "member")
       end
       redirect_to "/settings/organization",
                   notice: "You have joined the #{@organization.name} organization."
     else
-      not_found
+      flash[:error] = "The given organization secret was invalid."
+      redirect_to "/settings/organization"
```

Comment:
I used `redirect_to` instead of `render`  because rendering would require getting all the instance variables, and also for some reasons the styles weren't loaded with  `render`.